logging/logrus/options.go

```diff
@@ -118,3 +125,23 @@ func DefaultTripperwareCodeToLevel(httpStatusCode int) logrus.Level {
 		return logrus.WarnLevel
 	}
 }
+
+// WithRequestFieldExtractor adds a field, allowing you to customize what fields get prepopulated from the request.
+func WithRequestFieldExtractor(f RequestFieldExtractorFunc) Option {
+	return func(o *options) {
+		o.requestFieldExtractor = f
+	}
+}
+
+// WithRequestFieldExtractor adds a field, allowing you to customize what fields get prepopulated from the request.
+func WithResponseFieldExtractor(f ResponseFieldExtractorFunc) Option {
+	return func(o *options) {
+		o.responseFieldExtractor = f
+	}
+}
+
+// RequestFieldExtractorFunc is a signature of user-customizeable functions for extracting log fields from requests.
+type RequestFieldExtractorFunc func(req *http.Request) map[string]interface{}
+
+// ResponseFieldExtractorFunc is a signature of user-customizeable functions for extracting log fields from responses.
+type ResponseFieldExtractorFunc func(res httpwares.WrappedResponseWriter, req *http.Request) map[string]interface{}
```

Comment:
Why do you also pass in the `req` for the response field extractor

Reply:
We could get away with just passing the `ctx` at the moment as thats what we need to load the logger entry and then add new log fields.

Reply:
Actually did this initially due to some weirdness that we were doing internally.